File: BackEnd/WebServer/WebServer/Models/Api/Response/PostResponseModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using WebServer.Models.Database;
using WebServer.Models.Enums;

namespace WebServer.Models.Api.Response
{
    public class PostResponseModel
    {
        public int PostID { get; set; }

        public string Title { get; set; }

        public string Description { get; set; }

        public string ProfileImageID { get; set; }

        public string Username { get; set; }

        public int Likes { get; set; }

        public DateTime Date { get; set; }

        public ICollection<string> Images { get; set; }

        public ICollection<ActivityResponse> Activities { get; set; }

        public ICollection<CommentResponse> Comments { get; set; }
    }
}

File: FrontEnd/DemoApp/app/src/main/java/com/example/demoapp/data/model/api/response/PostResponseModel.java
package com.example.demoapp.data.model.api.response;

import com.example.demoapp.data.model.Activity;
import com.example.demoapp.data.model.Country;

import java.util.Date;
import java.util.List;

public class PostResponseModel
{
    private int postID;
    private String title;
    private String description;
    private Date date;
    private String username;
    private String profileImageID;
    private List<String> images;
    private List<Activity> activities;

    public String getTitle()
    {
        return title;
    }

    public void setTitle(String title)
    {
        this.title = title;
    }

    public String getDescription()
    {
        return description;
    }

    public void setDescription(String description)
    {
        this.description = description;
    }

    public Date getDate()
    {
        return date;
    }

    public void setDate(Date date)
    {
        this.date = date;
    }

    public String getUsername()
    {
        return username;
    }

    public void setUsername(String username)
    {
        this.username = username;
    }

    public String getProfileImageID()
    {
        return profileImageID;
    }

    public void setProfileImageID(String profileImageID)
    {
        this.profileImageID = profileImageID;
    }

    public List<String> getImages()
    {
        return images;
    }

    public void setImages(List<String> images)
    {
        this.images = images;
    }

    public List<Activity> getActivities()
    {
        return activities;
    }

    public void setActivities(List<Activity> activities)
    {
        this.activities = activities;
    }
}
